### PR TITLE
#87 - package hash to index.html

### LINKED ISSUE #87: Include hash in packages URL in index.html
Let's add hash to packages URL on index.html as requested by [simple repository API](https://www.python.org/dev/peps/pep-0503/):

> The URL SHOULD include a hash in the form of a URL fragment with the following syntax: #<hashname>=<hashvalue>, where <hashname> is the lowercase name of the hash function (such as sha256) and <hashvalue> is the hex encoded digest.
Repositories SHOULD choose a hash function from one of the ones guaranteed to be available via the hashlib module in the Python standard library (currently md5, sha1, sha224, sha256, sha384, sha512). The current recommendation is to use sha256.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Added packages sha256 digest hex to index.html to close #87.

### CODE REVIEW

File: src/main/java/com/artipie/pypi/http/SliceIndex.java [hunk 1]
```diff
@@ -79,26 +82,32 @@ public Response response(
...
-                                    "<a href=\"%s\">%s</a><br/>",
-                                    String.format("%s/%s", prefix, key.string()),
-                                    new KeyLastPart(key).get()
-                                )
+            SingleInterop.fromFuture(this.storage.list(rqkey))
+                .flatMapPublisher(Flowable::fromIterable)
+                .flatMapSingle(
+                    key -> Single.fromFuture(
+                        this.storage.value(key).thenCompose(
+                            value -> new ContentDigest(value, ContentDigest.Digests.SHA256).hex()
```
Comment (olegmoz) [inline src/main/java/com/artipie/pypi/http/SliceIndex.java:90]: @olenagerasimova please note, that reading all values from storage might be very inefficient in terms of resources. If query comes for too node we are going to read whole storage. I'd suggest to add and prioritize issue for calculating and storing hex values on adding package.

Comment (olenagerasimova) [reply src/main/java/com/artipie/pypi/http/SliceIndex.java:90]: @olegmoz thanks for noticing, I completely agree and planing to deal with `SliceIndex` performance later

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)[/z](https://www.0crat.com/u/carlosmiranda) didn't make CR comments

**Comment by 0crat** [comment]
Job was finished in 1 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The project doesn't have enough funds, can't make a payment

**Pull request merged**